util/fibers/detail/scheduler.cc

```diff
@@ -662,6 +673,46 @@ void Scheduler::RunDeferred() {
 #endif
 }
 
+void Scheduler::PrintAllFiberStackTraces() {
+  ProcessRemoteReady();
+
+  auto print_stack_trace = [](FiberInterface& fi) { fi.PrintStackTrace(); };
+
+  if (!ready_queue_.empty()) {
+    LOG(INFO) << "Ready fibers:";
+    for (auto& fiber : ready_queue_) {
+      print_stack_trace(fiber);
+    }
+  }
+  if (!terminate_queue_.empty()) {
+    LOG(INFO) << "Terminated fibers:";
+    for (auto& fiber : terminate_queue_) {
+      print_stack_trace(fiber);
+    }
+  }
+
+  bool print_sleeping_header = true;
+  if (!sleep_queue_.empty()) {
+    LOG(INFO) << "Sleeping fibers:";
+    print_sleeping_header = false;
+    for (auto& fiber : sleep_queue_) {
+      print_stack_trace(fiber);
+    }
+  }
+
+  // Print any other fibers that are sleeping outside of the scheduler's reach.
+  // Don't print ourselves?
+  for (auto& fiber : fibers_) {
```

Comment:
you do you need so many loops if you have a single list covering all the fibers. Is not it simpler to go over only "fibers_" and deduce fibers state based on their hooks?